### PR TITLE
Refactor CatAE

### PR DESCRIPTION
**Pull request by hakunanatasha**
CatAE refactor; seeding is taken care of in the overall code, but we may want to allow it as a kwarg.

### CODE REVIEW

File: lightwood/encoder/categorical/autoencoder.py [hunk 1]
```diff
@@ -10,98 +10,198 @@
...
+        self.onehot_encoder.prepare(priming_data)
+        self.batch_size = min(self.batch_size, int(len(priming_data) / 50))
+
+        train_loader = DataLoader(
+            list(zip(priming_data, priming_data)),
+            batch_size=self.batch_size,
+            shuffle=True,
+        )
+
+        # TODO; make `Gym` compatible with a dev set
```
Comment (paxcema) [inline lightwood/encoder/categorical/autoencoder.py:159]: Hmh, will turn this into an issue.

Comment (paxcema) [reply lightwood/encoder/categorical/autoencoder.py:159]: Wait, am now a bit confused. Gym supports a `test_data_loader`, which I would think works for this case. @George3d6 thoughts?

**Comment by George3d6** [comment]
I think this PR removed Gym completely @paxcema , which is amazing as we never use it, I will take this one over and try to include the removal of Gym as part of this.

**Comment by George3d6** [comment]
nvm, gym is not removed.

I think I will take over the CAE PR, remove the Gym and review at the same time since I'm not 100% clear on everything that's happening in the refactor.

**Comment by paxcema** [comment]
Any update on this one @George3d6?

**Comment by paxcema** [comment]
@hakunanatasha long overdue, merging and checking performance when next release comes around 👀

**Pull request merged**